src/main/kotlin/com/hypto/iam/server/apis/TokenApi.kt

```diff
@@ -59,8 +60,18 @@ suspend fun generateTokenOauth(call: ApplicationCall, context: ApplicationCall)
     val principal = context.principal<OAuthUserPrincipal>()!!
     val responseContentType = context.request.accept()
     val user = UserRepo.findByEmail(principal.email) ?: throw AuthenticationException("User has not signed up yet")
+    val userAuth = UserAuthRepo.fetchByUserHrnAndProviderName(user.hrn, principal.issuer)
+        ?: throw AuthenticationException("User has not signed up yet")
     val response = tokenService.generateJwtToken(ResourceHrn(user.hrn))
 
+    if (principal.metadata != null) {
+        if (userAuth.authMetadata == null) {
+            userAuthRepo.updateAuthMetadata(userAuth, principal.metadata)
+        } else if (userAuth.authMetadata != JSONB.valueOf(principal.metadata.toString())) {
```

Comment:
Can we compare only the required fields instead of the entire payload. Because, later if we store some metadata which we don't return in principal / token, things might break

Reply:
Created `authenticate` function to do the above, this results in empty function for `GoogleAuthProvider`